at_secondary/at_secondary_server/lib/src/verb/handler/scan_verb_handler.dart

```diff
@@ -133,17 +149,66 @@ class ScanVerbHandler extends AbstractVerbHandler {
                     .startsWith('${atConnectionMetadata.fromAtSign}:') ==
                 false) ||
             test.toString().startsWith('public:_'));
-        keyString = keys
-            .toString()
+        // keyString = keys
+        //     .toString()
+        //     .replaceAll('${atConnectionMetadata.fromAtSign}:', '');
+        keyString = jsonEncode(keys)
             .replaceAll('${atConnectionMetadata.fromAtSign}:', '');
       } else {
         // When pol is not performed, display only public keys
         keys.removeWhere((test) =>
             test.toString().startsWith('public:_') ||
             !test.toString().startsWith('public:'));
-        keyString = keys.toString().replaceAll('public:', '');
+        // keyString = keys.toString().replaceAll('public:', '');
+        keyString = jsonEncode(keys).replaceAll('public:', '');
       }
     }
+    print('keys type : ${keys.runtimeType}');
     return keyString;
   }
+
+  String? _prepareExternalKeysResponse(int page, String? keyString) {
+    if (keyString == null || keyString.isEmpty) {
+      return keyString;
+    }
+    var result = <String, dynamic>{};
+    print('keysString in _prepareExternalKeysResponse: $keyString');
+    keyString = keyString.replaceFirst('data:', '');
+    var keys = jsonDecode(keyString);
+    var start_index = (page - 1) * 10;
+    if (start_index < keys.length) {
+      var end_index = (start_index + 10 > keys.length)
+          ? keys.length - 1
+          : (start_index + 10);
+      result['keys'] = jsonEncode(keys.sublist(start_index, end_index));
+    } else {
+      result['keys'] = jsonEncode([]);
+    }
+    result['totalPages'] =
+        ((keys.length) / 10).toInt() + ((keys.length) % 10 > 0 ? 1 : 0);
+    result['keysPerPage'] = 10;
+    result['currentPage'] = page;
+    return jsonEncode(result);
+  }
+
+  String? _prepareLocalKeysResponse(int page, List<dynamic>? keys) {
```

Comment:
_prepareLocalKeysResponse and _prepareExternalKeysResponse have similar logic. This needs to be refactored to a single method called _preparePaginatedResponse and reuse.